### PR TITLE
Feature/card detail

### PR DESCRIPTION
**Pull request by iigorcunha**
### What this PR does (required):
- Create edit for card
- move routes endpoint to an index file inside of routes folder to handle the endpoints.
- split validate file into validations folder with each file to each controller.
- create new validations to validate request data.
- refactor controllers error handler.

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

https://user-images.githubusercontent.com/52304812/128405342-65e1d6ef-c083-4470-b827-aa3278cd86f1.mov

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

**Comment by iigorcunha** [comment]
#2

File: server/controllers/card.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      description,
+      colorCode,
+      deadline,
+      tags,
+      attachments
+    });
+  } catch (err) {
+    res.status(404)
+    throw new Error("No card found with given id")
+  }
```
Comment (moffatethan) [inline server/controllers/card.js:32]: So, async-express-handler actually handles catching any errors

Comment (moffatethan) [reply server/controllers/card.js:32]: Also, if no card is found then it will return null and not throw an error.

**Review by moffatethan — COMMENTED** [review]
Left some comments to address. Good work!

File: server/controllers/card.js (hunk 1 above)
Comment (iigorcunha) [reply server/controllers/card.js]: Yes, I saw that when I refactor the card delete. but I forgot to fix on cardEdit. thanks @moffatethan  I already fixed this issue.

**Review by moffatethan — APPROVED** [review]

**Comment by moffatethan** [comment]
Please address the merge conflicts @iigorcunha before merging 😄

**Pull request merged**